### ISSUE TITLE
Replace vm tests with native tests

### ISSUE THREAD

**Issue by srid** [juspay/services-flake#32]
Unlike NixOS services, process-compose programs created by services-flake do not mutate outside of `dataDir`, so we gain very little from the isolation provided by VMs.

Let's switch to running these tests natively, which gets us macOS support (resolving #14), in addition to allowing us to use Github Actions for CI (#11).

## Implementation idea

Use process-compose itself to run tests. process-compose running another process-compose, with readiness checks triggering the test process. We can use bash, or better nushell, to write the tests in. Nushell, because it can easily process the JSON from process-compose API.

**Comment by srid** [comment]
Note: this is mostly upstream work https://github.com/Platonic-Systems/process-compose-flake

**Comment by shivaraj-bh** [comment]
### Another approach

`testScript` can itself be a service that depends on the service that it testing. The `testScript` service should only start in CI.

**Comment by adrian-gierakowski** [comment]
> testScript can itself be a service that depends on the service that it testing. The testScript service should only start in CI.

Good idea! Once this [PR](https://github.com/F1bonacc1/process-compose/pull/78) is merged it will be super easy to run tests within process compose: once test process completes, other processes will be stopped and PC will exit with the code returned by the test process.

**Comment by shivaraj-bh** [comment]
I have been thinking of different ways to implement this. 

I have an idea and that involves having a `testScript` option for every service (resolves #33).
```nix
{
  testScript = lib.mkOption {
    type = types.nullOr (types.functionTo types.package);
    default = null;
    description = ''
      A package that contains a script to test the database.
      The script will be run after the database has been initialized.
    '';
    example = lib.literalExpression
    ''
      package: pkgs.writeShellApplication {
        name = "test-postgres";
        text = "echo 'SELECT version();' | ${package} -h 127.0.0.1";
      };
    '';
  };
}
```
The option will be used to create a process that will execute the `testScript` and end with whatever exit code the script returns (courtesy `exit_on_end`).
```nix
{
  "${name}-test" = lib.mkIf (config.testScript != null) {
    command = config.testScript config.package;
    depends_on."${name}".condition = "process_healthy";
    availability.exit_on_end = true;
  };
}
```

All looks good until we have more than one process that has a `testScript`, in that case whichever script executes first will be ran and all the other processes are immediately shutdown because `exit_on_end` waits for no one. I was thinking of a possible way to go around this by having a parent process that will use `exit_on_end` and this process will depend on all the `testScripts`. The parent process will have to somehow return the exit code 0 if all the `testScripts` ran successfully or the non-zero exit code of the first process that fails. I can't think of any way this can be done, any suggestions?
